autoload/neomake/makers/ft/elixir.vim

```diff
@@ -1,12 +1,28 @@
 " vim: ts=4 sw=4 et
 
+let g:neomake_elixir_credo_type_map= get(g:, 'neomake_elixir_credo_type_map', {'F': 'W','C': 'W','D': 'I','R': 'I'})
+
 function! neomake#makers#ft#elixir#PostprocessEnforceMaxBufferLine(entry) abort
     let buffer_lines = str2nr(line('$'))
     if (buffer_lines < a:entry.lnum)
         let a:entry.lnum = buffer_lines
     endif
 endfunction
 
+" Credo error types
+" F -> Refactoring opportunities
+" W -> Warnings
+" C -> Convention violation
+" D -> Software design suggestions
+" R -> Readability suggestions
+function! neomake#makers#ft#elixir#PostprocessCredoErrorType(entry) abort
+    let type = toupper(a:entry.type)
+    let type_map = g:neomake_elixir_credo_type_map
+    if has_key(type_map, type)
+        let a:entry.type = type_map[type]
+    endif
```

Comment:
No worries.
I've mean that `map()` could be used here.. but it's more like using `get()` is appropriate:

```diff
diff --git i/autoload/neomake/makers/ft/elixir.vim w/autoload/neomake/makers/ft/elixir.vim
index 704143e..dbe4111 100644
--- i/autoload/neomake/makers/ft/elixir.vim
+++ w/autoload/neomake/makers/ft/elixir.vim
@@ -21,12 +21,9 @@ function! neomake#makers#ft#elixir#PostprocessEnforceMaxBufferLine(entry) abort
 endfunction
 
 function! neomake#makers#ft#elixir#PostprocessCredo(entry) abort
-    let type = toupper(a:entry.type)
     let type_map = extend(s:neomake_elixir_credo_type_map,
                 \ get(g:, 'neomake_elixir_credo_type_map', {}))
-    if has_key(type_map, type)
-        let a:entry.type = type_map[type]
-    endif
+    let a:entry.type = get(type_map, toupper(a:entry.type), a:entry.type)
 endfunction
 
 function! neomake#makers#ft#elixir#EnabledMakers() abort
```

I am not sure it's more readable though, and there are no tests - so let's skip it.